File: sdks/java/examples/BasicUsage.java
import io.darkstrata.credentialcheck.*;
import io.darkstrata.credentialcheck.exception.*;

/**
 * Basic usage example for the DarkStrata Credential Check SDK.
 */
public class BasicUsage {

    public static void main(String[] args) {
        // Create a client with your API key
        try (DarkStrataCredentialCheck client = new DarkStrataCredentialCheck(
                ClientOptions.builder("your-api-key").build()
        )) {
            // Check a single credential
            CheckResult result = client.check("user@example.com", "password123");

            if (result.isFound()) {
                System.out.println("WARNING: Credential found in breach database!");
            } else {
                System.out.println("Credential not found in any known breaches.");
            }

            // Print metadata
            CheckMetadata metadata = result.getMetadata();
            System.out.println("Hash prefix: " + metadata.getPrefix());
            System.out.println("Total results for prefix: " + metadata.getTotalResults());
            System.out.println("HMAC source: " + metadata.getHmacSource());
            System.out.println("Checked at: " + metadata.getCheckedAt());

        } catch (ValidationException e) {
            System.err.println("Validation error: " + e.getMessage());
            if (e.getField() != null) {
                System.err.println("Field: " + e.getField());
            }
        } catch (AuthenticationException e) {
            System.err.println("Authentication failed: " + e.getMessage());
        } catch (DarkStrataException e) {
            System.err.println("Error: " + e.getMessage());
            System.err.println("Retryable: " + e.isRetryable());
        }
    }
}

File: sdks/php/examples/basic_usage.php
<?php

require __DIR__ . '/../vendor/autoload.php';

use DarkStrata\CredentialCheck\Client;
use DarkStrata\CredentialCheck\Crypto;
use DarkStrata\CredentialCheck\Exception\DarkStrataException;

$client = new Client(['apiKey' => getenv('DARKSTRATA_API_KEY')]);

// email and password come from your login form and are only ever used here.
// 1. Hash locally: SHA-256 of "email:password". Plaintext never leaves this process.
$hash = Crypto::hashCredential($argv[1] ?? 'user@example.com', $argv[2] ?? 'password123');

try {
    // 2. Only the first 5 characters of the hash are sent; the full hash is compared locally.
    $result = $client->checkHash($hash);
    echo $result->found ? "WARNING: credential found in a breach\n" : "OK: not found in known breaches\n";
    echo "prefix={$result->prefix} candidates={$result->totalResults} cached=" . var_export($result->cachedResult, true) . "\n";
} catch (DarkStrataException $e) {
    fwrite(STDERR, $e->getMessage() . ($e->isRetryable() ? ' (retryable)' : '') . "\n");
    exit(1);
}
